lstchain/io/io.py

```diff
@@ -1264,37 +1261,32 @@ def remove_duplicated_events(data):
     data.remove_rows(remove_row_list)
 
 
-def parse_cfg_bytestring(bytestring):
-    """
-    Parse configuration as read by eventio
-    :param bytes bytestring: A ``Bytes`` object with configuration data for one parameter
-    :return: Tuple in form ``('parameter_name', 'value')``
-    """
-    line_decoded = bytestring.decode('utf-8').rstrip()
-    if 'ECHO' in line_decoded or '#' in line_decoded:
-        return None
-    line_list = line_decoded.split('%', 1)[0]  # drop comment
-    res = re.sub(' +', ' ', line_list).strip().split(' ', 1)  # remove extra whitespaces and split
-    return res[0].upper(), res[1]
-
-
 def extract_simulation_nsb(filename):
     """
     Get current run NSB from configuration in simtel file
     :param str filename: Input file name
-    :return array of `float` by tel_id: NSB rate
-    """
-    nsb = []
-    with EventIOFile(filename) as f:
-        for o in yield_all_subobjects(f, [History, HistoryConfig]):
-            if hasattr(o, 'parse'):
-                try:
-                    cfg_element = parse_cfg_bytestring(o.parse()[1])
-                    if cfg_element is not None:
-                        if cfg_element[0] == 'NIGHTSKY_BACKGROUND':
-                            nsb.append(float(cfg_element[1].strip('all:')))
-                except Exception as e:
-                    print('Unexpected end of %s,\n caught exception %s', filename, e)
+    :return dict of `float` by tel_id: NSB rate
+    """
+    nsb = {}
+    # In current MC, correct NSB are logged after 'STORE_PHOTOELECTRONS' entries
+    # TODO In any new production, behaviour needs to be verified.
+    # New version of simtel will allow to use better metadata
```

Comment:
Maybe this could be a Note in the function docstring instead of a comment here so it is visible e.g. in the documentation